### ISSUE TITLE
OBS 26 fails to switch to other Chrome windows after closing currently captured one

### ISSUE THREAD

**Issue by flaeri** [obsproject/obs-studio#3537]
## Platform
Operating system and version: Win10 2004 (19041)
OBS Studio version: 26
Chrome version: 85.0.4183.121

## Expected Behavior
When you are window capturing chrome, having match priority to "match title, otherwise same executable", and have multiple separate chrome windows open, I expect that when I would close the currently captured one, it would find the other existing ones as it did in 25.0.8.

## Current Behavior
OBS seems to be stuck on the first window, and refuses to switch to other windows of chrome after closing the original.

## Steps to Reproduce
1. Open up two Chrome windows. Have both visible and rendered on screen.
2. Add window capture to obs, select one of the chrome windows, and change the match priority to "match title, or other executable
3. Close the chrome window currently selected.
4. Observe as OBS never switches to the other window

## Additional information
Testing 25.0.8 on the same system, with the same profile/scene collection, this behavior is different. OBS switches to the other visible and active chrome window within a second or so.
This behavior seems to only apply to chrome so far. Firefox, edge (new chromium one), cmd and others seem to switch just fine, using the same steps.

Log: https://obsproject.com/logs/DiefvQhgIA353odx

**Comment by Gol-D-Ace** [comment]
Can confirm this with 26.0.2 on Windows 10 2004 Pro x64.

Some additional info / discoveries:

- Having just one Chrome window open, closing it and reopening it continues capturing fine here
- If both Chrome windows have the same website open (title) it will also continue to work and will switch like expected

**Comment by kutu** [comment]
Having the same issue
- Open obs
- capture chrome (same title, or same executable)
- close obs
- change tab in chrome (so it has different title)
- open obs
- capture failed

changing tabs while obs is opened everything is fine

was working on obs 25

**Comment by keirakatana** [comment]
I've tested this with:
Chrome
Edge
Brave
Firefox

The test was this. I open a new tab, set capture window and also selecting 'same executable'. I tell the browser to remember last page seen. Then I click on a Twitch page. I close the brower, then reopen it. Chrome, Edge, Brave all failed to capture until I opened a new tab. Firefox however worked, not matter what the tab was.

**Comment by flaeri** [comment]
I did experience the issues that @keirakatana was experiencing in the past, so I went on to verify, but I could not very in chrome, and to my surprise, on 26.1.1 the original post is no longer reproducible. Seems to me like this was fixed (at least for me).

@keirakatana please confirm OS (windows version if applicable), and OBS version.

@Gol-D-Ace Could you do a new check and confirm whether the original issue is still reproducible for you?

**Comment by keirakatana** [comment]
Windows 10 pro 20H2
OBS 26.1.1(64bit)

**Comment by Gol-D-Ace** [comment]
Did work fine for one try.
Tried it again and no switching.
26.1.1

**Comment by Durss** [comment]
Finally i find people having the same issue.
Here is another reproductible step by step for my case :

1. open chrome with 2 different tabs
2. configure window capture on chrome (the process will have the current tab's title on its name)
3. capture works OK
4. close the captured tab
5. OBS switches properly to the second tab (apparently this doesn't work if it's a different window as stated above tho)
6. close chrome
7. OBS capture becomes empty as expected
8. Open chrome back
9. OBS fails to attach to the process as the title is different than the one configured

I had the exact same issue with VSCode.
I managed to fix it by removing the dynamic window title that included the currently edited file's name.

TLDR : no matter what option you choose, the process' name seem crucial for OBS to properly capture it.
If we could avoid chrome process to change its name with the current tab's name i'm 99% sure we would have no issue.
Here i'm talking about chrome but this is the same for other browsers (firefox, brave, vivaldi, edge...) or any software that changes its title.

**Comment by lindenkron** [comment]
Confirmed still an issue.
Windows 10 19H2, 64 bit.
OBS 26.1.1

**Comment by kutu** [comment]
temporary solution: chrome 90 now allows to change its window name permanently, so it will not be a tab name
right click on arrow button -> "Name window..."
![image](https://user-images.githubusercontent.com/228802/115948586-ee715880-a4df-11eb-8120-ae2464b65476.png)

**Comment by LanceLake** [comment]
This does not help. If you rename the window, this name doesn't work if you open up a new tab or close chrome and reopen it.

**Comment by kutu** [comment]
works for me, i renamed it to "Chrome", and any new tab or when i reopen chrome, window still has name "Chrome"

**Comment by LanceLake** [comment]
Operating system? 

Can we see a recording of this happening? 
![Bug](https://user-images.githubusercontent.com/6218915/115949296-ea661c00-a488-11eb-9d58-6d5c1d59d8a8.gif)

**Comment by kutu** [comment]
i have an option set to "reopen last session", maybe thats why your chrome lose window name after restart

**Comment by keirakatana** [comment]
> i have an option set to "reopen last session", maybe thats why your chrome lose window name after restart

That's fine if the last session was your 'chrome' labelled tab. If you open a new tab on another page, then close the 'chrome' tab. Then your 'last session' won't included the tab labelled 'chrome'.

**Comment by Kenshin9977** [comment]
I have the same issue with Visual Studio Code. I don't even have to close OBS to reproduce the issue, just closing VSCode and reopening. It didn't work either on v25. Here are the exact steps :

1. Open OBS
2. Open Visual Studio Code
3. Capture VSCode (same title, or same executable)
4. Go to another tab within VSCode
5. Close VSCode
6. Open VSCode
7. OBS Fails to capture VSCode

In the meantime I'll be using Sublime Text or Atom.

**Comment by Durss** [comment]
> I have the same issue with Visual Studio Code. I don't even have to close OBS to reproduce the issue, just closing VSCode and reopening. It didn't work either on v25. It shares code from Chrome so the issue is surely  

You can fix the issue on VsCode by fixing the window's title. By default the title of VsCode contains the currently edited file, if you remove that in the preferences you won't have that problem anymore.

If we could do the same on chrome and other softwares having this issue the problem would probably be less annoying.

**Comment by ItsCubeTime** [comment]
Also experiencing this issue on Win11 OBS v26.1.1 (using a plugin that wont work on more recent versions of OBS, so I cant upgrade).

**Comment by mredodos** [comment]
> @Kenshin9977
> 
> > I have the same issue with Visual Studio Code. I don't even have to close OBS to reproduce the issue, just closing VSCode and reopening. It didn't work either on v25. It shares code from Chrome so the issue is surely
> 
> You can fix the issue on VsCode by fixing the window's title. By default the title of VsCode contains the currently edited file, if you remove that in the preferences you won't have that problem anymore.
> 
> If we could do the same on chrome and other softwares having this issue the problem would probably be less annoying.

that is not a fix is a work around. if i select same executable or same class theoretically i dont lost the source. but its not true if the title is similar or the same when i close the windows obs lost the source and not switch on 

Not work with visual studio for the same problem not work on chrome.

**Comment by OmgImAlexis** [comment]
I'm on Windows 11 with OBS 27.2.3 and I'm having the same issues.

For the three options.

1: If I close and reopen the exact same tab it can rehook the window.
2: Same as above.
3: If I close chrome and reopen OBS it never rehooks the window, I have to go in and change the window each time.

**Comment by mossyroo** [comment]
this still hasn't been solved as of May 2022, I wish there was a proper fix for it

**Comment by keirakatana** [comment]
It says I have to log in and I have no record of my account, that's why
I've requested this.

**Comment by LanceLake** [comment]
We can't unsubscribe you. Try to log in and click forgot password.

**Comment by flaeri** [comment]
Can no longer reproduce using my own repro steps.

If others feel different, please report with what steps you need to do in order to reproduce.
